tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/docerina/FieldLevelDocsTest.java

```diff
@@ -0,0 +1,235 @@
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.test.docerina;
+
+import org.ballerinalang.docgen.docs.BallerinaDocGenerator;
+import org.ballerinalang.docgen.generator.model.DefaultableVarible;
+import org.ballerinalang.docgen.generator.model.Module;
+import org.ballerinalang.docgen.generator.model.Object;
+import org.ballerinalang.docgen.generator.model.Project;
+import org.ballerinalang.docgen.generator.model.Record;
+import org.ballerinalang.docgen.model.ModuleDoc;
+import org.ballerinalang.test.util.BCompileUtil;
+import org.ballerinalang.test.util.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test class to check record/object field-level documentation in docerina.
+ */
+public class FieldLevelDocsTest {
+
+    private Module testModule;
+    // record with module-level documentation
+    private Record addressRecord;
+    // record with field-level documentation
+    private Record personRecord;
+    // record with both module-level & field-level documentation
+    private Record apartmentRecord;
+    // Object with module-level documentation
+    private Object studentObj;
+    // Object with field-level documentation
+    private Object teacherObj;
+
+    private static final String PARAGRAPH_OPEN_TAG = "<p>";
+    private static final String PARAGRAPH_CLOSE_TAG = "</p>";
+    private static final String NEW_LINE = "\n";
+    private static final String PARAGRAPH_CLOSE_WITH_NEW_LINE = PARAGRAPH_CLOSE_TAG + NEW_LINE;
+
+    @BeforeClass
+    public void setup() throws IOException {
+        String sourceRoot = "test-src/docerina/fieldLevelDocsProject";
+        CompileResult result = BCompileUtil.compile(sourceRoot, "testModule");
+
+        List<BLangPackage> modules = new LinkedList<>();
+        modules.add((BLangPackage) result.getAST());
+        Map<String, ModuleDoc> docsMap = BallerinaDocGenerator.generateModuleDocsFromBLangPackages(
+                Paths.get("src/test/resources", sourceRoot).toAbsolutePath().toString(), modules);
+        List<ModuleDoc> moduleDocList = new ArrayList<>(docsMap.values());
+        moduleDocList.sort(Comparator.comparing(pkg -> pkg.bLangPackage.packageID.toString()));
+        Map<String, List<Path>> resources = new HashMap<>();
+
+        Project project = BallerinaDocGenerator.getDocsGenModel(moduleDocList, resources);
+        testModule = project.modules.get(0);
+
+        List<Record> records = testModule.records;
```

Comment:
@pramodya1994 Do we need to declare these variable if its only used once?

Reply:
Removed